js.go

```diff
@@ -2047,7 +2051,26 @@ func (sub *Subscription) resetOrderedConsumer(sseq uint64) {
 		js := jsi.js
 		sub.mu.Unlock()
 
-		consName := nuid.Next()
+		// Attempt to delete the existing consumer.
+		// If ErrConsumerNotFound is returned, it means that the consumer was already deleted
+		// by the server, so we can proceed to create a new consumer.
+		err := sub.deleteConsumer()
+		if err != nil && !errors.Is(err, ErrConsumerNotFound) {
+			var apiErr *APIError
+			if errors.Is(err, ErrJetStreamNotEnabled) || errors.Is(err, ErrTimeout) || errors.Is(err, context.DeadlineExceeded) {
+				// if creating consumer failed, retry
```

Comment:
here we abort recreating the ordered consumer if deleting the previous consumer fails? I wonder if we should ignore the delete consumer error but call `pushErr` here and then proceed to recreate the ordered consumer instead as before

Reply:
The delete call should be async and best effort, I would not even wait for a response.

Reply:
Good points, done